### PR TITLE
Journal the whole test fixture method, not just the OperationContract…

### PR DESCRIPTION
**Pull request by ewiseblatt**
@jtk54

### CODE REVIEW

File: citest/base/base_test_case.py [hunk 1]
```diff
@@ -51,29 +59,119 @@ class BaseTestCase(unittest.TestCase):
...
+        if self.__in_step == _TestProcessingStep.SETUP
+        else 'tearDown')
+
+  def __end_step_context(self, relation):
+    if self.__in_step is None:
+      return
+    JournalLogger.end_context(relation=relation)
+
+  def __call__(self, result=None):
+    """Wraps the base class fixture heuristics that run an individual test."""
```
Comment (jtk54) [inline citest/base/base_test_case.py]: Hueristics?

File: citest/reporting/html_renderer.py [hunk 2]
```diff
@@ -528,35 +528,56 @@ def terminate(self):
...
+                         title_html + ' (<i>empty</i>)', css=css)
+    else:
+      delta_time = end_control['_timestamp'] - begin_control['_timestamp']
+      if css:
+        title_html = '<span{css}>{title}</span>'.format(
+            css=css, title=title_html)
+      lvl = min(1, len(self.__context_stack))
+      title_html = '<context{n}>{title}</context{n}>'.format(
+          n=lvl, title=title_html)
+      html = '{title}\n<table>\n{rows}\n</table>\n'.format(
```
Comment (jtk54) [inline citest/reporting/html_renderer.py]: This is brutal, considering things like `beautifulsoup`, etc exist.

File: citest/reporting/html_renderer.py [hunk 3]
```diff
@@ -528,35 +528,56 @@ def terminate(self):
...
-              css=css, title=title_html)
-        lvl = min(1, len(self.__context_stack))
-        title_html = '<context{n}>{title}</context{n}>'.format(
-            n=lvl, title=title_html)
-        html = '{title}\n<table>\n{rows}\n</table>\n'.format(
-            title=title_html, rows=''.join(held.html))
-        context_title = '%s <small>+%.3fs</small>' % (title_html, delta_time)
-        self.render_log_tr(held.control['_timestamp'], context_title, html)
+      if (not just_finished.html
+          and just_finished.control.get('_title', '') in ['setUp', 'tearDown']):
```
Comment (jtk54) [inline citest/reporting/html_renderer.py]: Is this `setUp` and `tearDown` for a single test case? I thought those applied to the entire fixture, not the individual cases?

File: citest/base/base_test_case.py (hunk 1 above)
Comment (ewiseblatt) [inline citest/base/base_test_case.py]: heuristics

File: citest/reporting/html_renderer.py (hunk 3 above)
Comment (ewiseblatt) [inline citest/reporting/html_renderer.py]: setUpTestCase/tearDownTestCase are class methods, setUp/tearDown is an instance method. The instance methods are called before each individual test. The class methods are called before the first and after the last test from a given fixture.

Comment (jtk54) [inline citest/reporting/html_renderer.py]: Ok got it.

**Comment by jtk54** [comment]
LGTM

**Pull request merged**